aspnetcore/fundamentals/static-files/samples/6.x/StaticFilesSample/Program.cs

```diff
@@ -385,7 +385,51 @@
 app.UseStaticFiles(new StaticFileOptions
 {
     FileProvider = new PhysicalFileProvider(
-        Path.Combine(builder.Environment.ContentRootPath, "static"))
+        Path.Combine(builder.Environment.ContentRootPath, "MyStaticFiles"))
+});
+#endregion
+
+app.UseAuthorization();
+
+app.MapDefaultControllerRoute();
+app.MapRazorPages();
+
+app.Run();
+#elif MULT2
+using Microsoft.Extensions.FileProviders;
+
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddRazorPages();
+builder.Services.AddControllersWithViews();
+
+var app = builder.Build();
+
+if (!app.Environment.IsDevelopment())
+{
+    app.UseExceptionHandler("/Error");
+    app.UseHsts();
+}
+
+app.UseHttpsRedirection();
+
+#region snippet_mult2
+var webRootProvider = new PhysicalFileProvider(builder.Environment.WebRootPath);
+var newPathProvider = new PhysicalFileProvider(
+  Path.Combine(builder.Environment.ContentRootPath, "MyStaticFiles"));
+
+var compositeProvider = new CompositeFileProvider(webRootProvider,
+                                                  newPathProvider);
+
+// Update the default provider.
+app.Environment.WebRootFileProvider = compositeProvider;
+
+app.UseStaticFiles();
+
+app.UseStaticFiles(new StaticFileOptions()
+{
+    FileProvider = new PhysicalFileProvider(
+             Path.Combine(builder.Environment.ContentRootPath, "MyStaticFiles"))
 });
```

Comment:
You shouldn't need a second call to UseStaticFiles, the first one will look at WebRootFileProvider and check both locations from the CompositeFileProvider.